GeoLib/PointVec.h

```diff
@@ -143,6 +145,10 @@ class PointVec : public TemplateVec<Point>
 	 */
 	std::vector<std::size_t> _pnt_id_map;
 
+	/// The reverse map to the name to id map, for fast lookup of the name to a
+	/// given point id.
+	std::vector<std::string> _id_to_name_map;
```

Comment:
for me it looks waste of memory because often we assign names to a few number of points. how about storing both string and point id as a lookup table like `std::vector<std::set<std::string, std::size_t>>` or `std::map<std::string, std::size_t>`?